src/Uno.UI/UI/Xaml/Controls/TextBlock/JavaStringCache.android.cs

```diff
@@ -0,0 +1,189 @@
+﻿#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading;
+using System.Text;
+using Windows.Foundation;
+
+using Uno;
+using Uno.Extensions;
+using Uno.UI;
+using Uno.Foundation.Logging;
+using Windows.UI.Xaml.Media;
+using Uno.Collections;
+using Android.Security.Keystore;
+using Java.Security;
+using Uno.Buffers;
+using Windows.System;
+
+namespace Windows.UI.Xaml.Controls
+{
+	/// <summary>
+	/// A TextBlock measure cache for non-formatted text.
+	/// </summary>
+	internal static class JavaStringCache
+	{
+		private static Logger _log = typeof(JavaStringCache).Log();
+		private static Stopwatch _watch = Stopwatch.StartNew();
+		private static HashtableEx _table = new();
+		private static TimeSpan _lastScavenge;
+		private static object _gate = new();
+
+		internal static readonly TimeSpan LowMemoryTrimInterval = TimeSpan.FromMinutes(5);
+		internal static readonly TimeSpan MediumMemoryTrimInterval = TimeSpan.FromMinutes(3);
+		internal static readonly TimeSpan HighMemoryTrimInterval = TimeSpan.FromMinutes(1);
+		internal static readonly TimeSpan OverLimitMemoryTrimInterval = TimeSpan.FromMinutes(.5);
+
+		internal static readonly TimeSpan ScavengeInterval = TimeSpan.FromMinutes(.5);
+
+		private static DefaultArrayPoolPlatformProvider _platformProvider = new DefaultArrayPoolPlatformProvider();
+
+		/// <summary>Determines if automatic memory management is enabled</summary>
+		private static readonly bool _automaticManagement;
+		/// <summary>Determines if GC trim callback has been registerd if non-zero</summary>
+		private static int _trimCallbackCreated;
+
+		private record KeyEntry(string Value, Java.Lang.String NativeValue)
+		{
+			public TimeSpan LastUse { get; set; } = _watch.Elapsed;
+		}
+
+		static JavaStringCache()
+		{
+			_automaticManagement = WinRTFeatureConfiguration.ArrayPool.EnableAutomaticMemoryManagement && _platformProvider.CanUseMemoryManager;
```

Comment:
It it possible to assign this inline and avoid the static constructor?

Reply:
> but the static constructor will still be here

Why does it need to stay? Wouldn't it be empty if we inlined this assignment?

Reply:
Visually yes, but in IL it won't change much, the .cctor will still be generated.

Reply:
Another thing here is the fact that ordering may become a problem, since this particular line depends on an inline initialized member, so I think it's best to keep it this way.

Reply:
@jeromelaban Yes it is still generated, but we lost `beforefieldinit` (https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1810 has some details).

The performance difference might be too small, so it doesn't matter that much.